table/block_based/block_based_table_reader.cc

```diff
@@ -2581,6 +2580,10 @@ void BlockBasedTable::MultiGet(const ReadOptions& read_options,
                 read_options, results[idx_in_batch], &first_biter,
                 statuses[idx_in_batch]);
             reusing_block = false;
+          } else {
+            // If handler is null and result is empty, then the status is never
+            // set, which should be the initial value: ok().
+            assert(statuses[idx_in_batch].ok());
```

Comment:
Oh interesting. Is it a better fix than https://github.com/facebook/rocksdb/blob/master/table/block_based/block_based_table_reader.cc#L1695-L1697  ? CC @anand1976

Reply:
It does look more elegant, so I'd go with this instead of https://github.com/facebook/rocksdb/blob/master/table/block_based/block_based_table_reader.cc#L2714-L2719.

I wonder if a default initialized ```Status``` should simply not enforce the check. Cc @ajkr